pkg/mcs/tso/server/server.go

```diff
@@ -250,8 +430,155 @@ func checkStream(streamCtx context.Context, cancel context.CancelFunc, done chan
 }
 
 // GetTLSConfig get the security config.
-// TODO: implement it
 func (s *Server) GetTLSConfig() *grpcutil.TLSConfig {
+	return &s.cfg.Security.TLSConfig
+}
+
+func (s *Server) initClient() error {
+	tlsConfig, err := s.cfg.Security.ToTLSConfig()
+	if err != nil {
+		return err
+	}
+	u, err := types.NewURLs(strings.Split(s.cfg.BackendEndpoints, ","))
+	if err != nil {
+		return err
+	}
+	s.backendUrls = []url.URL(u)
+	s.etcdClient, s.httpClient, err = etcdutil.CreateClients(tlsConfig, s.backendUrls)
+	return err
+}
+
+func (s *Server) startGRPCServer(l net.Listener) {
+	defer s.serverLoopWg.Done()
+
+	gs := grpc.NewServer()
+	s.service.RegisterGRPCService(gs)
+	serverr := gs.Serve(l)
+
+	log.Info("grpc server stopped serving.", errs.ZapError(serverr))
+	log.Info("try to gracefully stop the server now.")
+
+	// Attempt graceful stop (waits for pending RPCs), but force a stop if
+	// it doesn't happen in a reasonable amount of time.
+	done := make(chan struct{})
+	go func() {
+		gs.GracefulStop()
+		close(done)
+	}()
+	select {
+	case <-done:
+	case <-time.After(defaultGRPCGracefulStopTimeout):
+		log.Info("stopping grpc gracefully is taking longer than expected and force stopping now.")
+		gs.Stop()
+	}
+
+	if s.IsClosed() {
+		log.Info("grpc server stopped.")
+	} else {
+		log.Fatal("grpc server stopped unexpectedly.", errs.ZapError(serverr))
```

Comment:
the error is the same with L458. I think we don't need to output error in L458 when expectly

Reply:
I need L458 since I want to record and monitor the error code even when IsClosed() is true. How about changing it to "log.Info("grpc server stopped serving. try to gracefully stop the server now", errs.ZapError(serverr))" by merging L458 and L459

Reply:
removed the error code when exit normally